crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -103,6 +103,8 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
             .map(ForeignInto::foreign_into)
             .or(payment_intent.setup_future_usage);
 
+        payment_intent.udf = payment_intent.udf.or(request.udf.clone());
```

Comment:
If the merchant hasn't specified udf when he created intent,
we shouldn't be picking one up from the confirm request as well right?

Reply:
I feel the same But the task was to add udf to PaymentsRequest and whichever API's using it.